modules/capture-admin-service-impl/src/main/java/org/opencastproject/capture/admin/impl/CaptureAgentStateServiceImpl.java

```diff
@@ -198,19 +206,19 @@ protected AgentImpl getAgent(String name, String org) throws NotFoundException {
    *          the unique agent name
    * @param organization
    *          the organization
-   * @param em
-   *          the entity manager
    * @return the agent or <code>null</code> if no agent has been found
    */
-  protected AgentImpl getAgentEntity(String name, String organization, EntityManager em) throws NotFoundException {
-    try {
-      Query q = em.createNamedQuery("Agent.get");
-      q.setParameter("id", name);
-      q.setParameter("org", organization);
-      return (AgentImpl) q.getSingleResult();
-    } catch (NoResultException e) {
-      throw new NotFoundException(e);
-    }
+  protected ThrowingFunction<EntityManager, AgentImpl, NotFoundException> getAgentEntityQuery(String name, String organization) {
+    return em -> {
+      try {
+        TypedQuery<AgentImpl> q = em.createNamedQuery("Agent.get", AgentImpl.class);
```

Comment:
What's the advantage of this compared to

```java
  protected AgentImpl getAgentEntityQuery(EntityManager em, String name, String organization) {
    try {
      TypedQuery<AgentImpl> q = em.createNamedQuery("Agent.get", AgentImpl.class);
      …
```

The latter form is far more common in the Java world and will probably mean that new developers are more comfortable with the code, especially if you then combine it with calls like this where I'm wondering why we would construct this function to just call the function?

```java
AgentImpl existing = getAgentEntityQuery(agent.getName(), agent.getOrganization()).apply(em);
```

Reply:
The idea is that you can directly pass this to `exec...` functions, e.g.

```java
AgentImpl existing = db.exec(getAgentEntityQuery(String name, String organization));
```

In more complex scenarios with multiple queries using the same `EntityManager` (e.g. a larger transactions), this is a little bit awkward, but I wanted to keep it consistent.

Reply:
The idea is that you can directly pass this to `exec...` functions, e.g.

```java
AgentImpl existing = db.exec(getAgentEntityQuery(String name, String organization));
```

In more complex scenarios with multiple queries using the same `EntityManager` (e.g. a larger transactions), this is a little bit awkward, but I wanted to keep it consistent.